core/src/main/scala/org/apache/spark/SparkEnv.scala

```diff
@@ -442,9 +445,13 @@ object SparkEnv extends Logging {
     val addedJarsAndFiles = (addedJars ++ addedFiles).map((_, "Added By User"))
     val classPaths = (addedJarsAndFiles ++ classPathEntries).sorted
 
+    // Add Hadoop properties, it will not ignore configs including in Spark. Some spark
+    // conf starting with "spark.hadoop" may overwrite it.
+    val hadoopProperties = hadoopConf.asScala.map(entry => (entry.getKey, entry.getValue)).toSeq
```

Comment:
I think `.asScala.toSeq` works directly too, no big deal

Reply:
`Seq[(String, String)] ` will not match `Seq[Map.Entry]` here. Maybe it will work when we use:

> val hadoopProperties = hadoopConf.asScala.toSeq.map(entry => (entry.getKey, entry.getValue))

Reply:
FWIW:
```
scala> val foo = new java.util.HashMap[String,String]()
scala> foo.put("foo", "bar")
scala> import scala.collection.JavaConverters._
scala> foo.asScala.toSeq
res1: Seq[(String, String)] = ArrayBuffer((foo,bar))
```